airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -59,3 +61,28 @@ def get_datasets(
     query = apply_sorting(query, order_by, {}, allowed_filter_attrs)
     datasets = query.offset(offset).limit(limit).all()
     return dataset_collection_schema.dump(DatasetCollection(datasets=datasets, total_entries=total_entries))
+
+
+@security.requires_access([(permissions.ACTION_CAN_READ, permissions.RESOURCE_DATASET)])
+@provide_session
+@format_parameters({'limit': check_limit})
+def get_dataset_events(
+    id, *, limit: int, offset: int = 0, order_by: str = "created_at", session: Session = NEW_SESSION
+):
+    """Get events for a Dataset"""
+    dataset = session.query(Dataset).get(id)
+    if not dataset:
+        raise NotFound(
+            "Dataset not found",
+            detail=f"The Dataset with id: `{id}` was not found",
+        )
+
+    allowed_attrs = ['task_id', 'dag_id', 'run_id', 'map_index', 'created_at']
```

Comment:
should we allow sorting by `map_index` alone? Practically, I feel like really this would at least combine with the task_id, right?

Reply:
yeah, this is fine for now. I just wanted to bring it up for later.